### PR TITLE
feat(cd): enable deployments via GitHub releases

### PR DESCRIPTION
**Pull request by Nitish145**
Describe the changes you have made in this PR -
- Uses GitHub releases & associated tags to ship the app to play store.
- gh-actions accepts 'vx.y.z-beta' for beta builds and 'vx.y.z' for prod builds.

Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

### CODE REVIEW

**Review by manjotsidhu — APPROVED** [review]

**Comment by manjotsidhu** [comment]
We can delete `deploy` branch after merging this since it will not be required. Also, a corner case would be where I can make release based off branch another than `master`. I feel we should only allow from `master` branch.

**Comment by tachyons** [comment]
https://github.com/xotahal/fastlane-plugin-semantic_release

**Comment by tachyons** [comment]
https://bumped.github.io/

**Comment by tachyons** [comment]
https://www.conventionalcommits.org/en/about/ There are some tools listed here, we should explore them as well to improve the workflow by making use of conventional commits structure we decided to follow

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -18,9 +15,17 @@ jobs:
...
+    # This job will run if the release is based on master branch
+    if: github.event.release.target_commitish == 'master'
     steps:
-      # Setup Java environment in order to build the Android app.
       - uses: actions/checkout@v1
+
+      # Extract the release type
+      - name: set release type
+        id: release
+        run: echo ::set-output name=type::${{ endsWith(github.event.release.tag_name, 'beta') && 'beta' || 'prod' }}
```
[the remark this answers is not in the document]
Comment (Nitish145) [reply .github/workflows/cd.yml]: Done!

**Comment by Nitish145** [comment]
> https://github.com/xotahal/fastlane-plugin-semantic_release

This looks great! @manjotsidhu would you want to take this ahead? Maybe in this PR or another one.

**Comment by manjotsidhu** [comment]
> > https://github.com/xotahal/fastlane-plugin-semantic_release
> 
> This looks great! @manjotsidhu would you want to take this ahead? Maybe in this PR or another one.

Sure, I think we can handle that in a new PR.

**Review by manjotsidhu — APPROVED** [review]

**Pull request merged**